metals/src/main/scala/scala/meta/internal/builds/BloopInstallProvider.scala

```diff
@@ -0,0 +1,27 @@
+package scala.meta.internal.builds
+import scala.concurrent.Future
+
+import scala.meta.internal.metals.MetalsLanguageClient
+import scala.meta.io.AbsolutePath
+
+/**
+ * Helper trait for build tools that have a Bloop plugin
+ */
+trait BloopInstallProvider { this: BuildTool =>
```

Comment:
This used to actually be the `BloopPluginBuildTool.scala`.  I've changed this to a trait mainly to allow it to be a mixing alongside of `BuildServerProvider.scala` for build tools that can also serve as build servers. My hope is that as we move forward to for example support Mill BSP, it will be much simpler to be able to just add the `BuildServerProvider` mixin to the `MillBuildTool`, add in the args for the bsp config generation, and that's literally almost all that would be needed.